theano/gpuarray/dnn_fwd.c

```diff
@@ -138,6 +141,16 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
           return 1;
         }
         algo = choice.algo;
+
+        #ifdef DEBUG
+        if (count == 0) {
+            PyErr_SetString(PyExc_RuntimeError, "No best-timed conv fwd algorithm found");
+            return 1;
+        } else {
+            fprintf(stderr, "(%d best-timed conv fwd algorithms)\n", count);
```

Comment:
Printing the content of the `cudnnConvolutionFwdAlgoPerf_t` that is returned in `choice` as well would be nice.
In fact, there should probably be an error if `algo.status != CUDNN_STATUS_SUCCESS`, like if `count == 0`.